nmostesting/TestHelper.py

```diff
@@ -266,10 +267,86 @@ def get_messages(self):
         return msg_cpy
 
     def did_error_occur(self):
-        return self.error_occured
+        return self.error_occurred
 
     def get_error_message(self):
         return self.error_message
 
     def clear_messages(self):
         self.messages.clear()
+
+
+class MQTTClientWorker:
+    """MQTT Client Worker"""
+    def __init__(self, host, port, secure=False, username=None, password=None):
+        """
+        Initializer
+        :param host: broker hostname (string)
+        :param port: broker port (int)
+        :param secure: use TLS (bool)
+        :param username: broker username (int)
+        :param password: broker password (int)
+        """
+        self.host = host
+        self.port = port
+        self.error_occurred = False
+        self.connected = False
+        self.error_message = ""
+        # MQTT 5 is required so that we can set retainAsPublished
+        # when subscribing to test whether messages have retain flags set
+        self.client = mqtt.Client(protocol=mqtt.MQTTv5)
+        self.client.on_connect = lambda client, userdata, flags, rc, properties=None: self.on_connect(flags, rc)
+        self.client.on_disconnect = lambda client, userdata, rc: self.on_disconnect(rc)
+        self.client.on_message = lambda client, userdata, msg: self.on_message(msg)
+        self.client.on_subscribe = lambda client, userdata, mid, *args: self.on_subscribe(mid)
+        self.client.on_log = lambda client, userdata, level, buf: self.on_log(level, buf)
+        if secure:
+            self.client.tls_set(CONFIG.CERT_TRUST_ROOT_CA)
+        if username or password:
+            self.client.username_pw_set(username, password)
+        self.messages = []
+
+    def start(self):
+        self.client.connect_async(self.host, self.port)
+        self.client.loop_start()
+
+    def close(self):
+        self.client.loop_stop()
+
+    def on_connect(self, flags, rc):
+        result, message_id = self.client.subscribe("x-nmos/#", options=mqtt.SubscribeOptions(retainAsPublished=True))

```

Comment:
Does it matter not all senders may use the recommended format beginning with `x-nmos`?